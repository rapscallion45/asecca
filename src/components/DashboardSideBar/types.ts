/*
 * Dashboard Sidebar type definitions
 *
 * These type definitions are used in the applications Dashboard Sidebar component.
 */

/**
 * Nav Bar sidebar on close callback function typedef
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 *
 * @callback IDashboardSidebarOnCloseCallback
 * @returns {void} - no return value
 */
export interface IDashboardSidebarOnCloseCallback {
  (): void;
}
