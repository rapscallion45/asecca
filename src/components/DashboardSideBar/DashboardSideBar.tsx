import { FC, useEffect } from 'react';
import Link from 'next/link';
import { useRouter } from 'next/router';
import { styled } from '@mui/material/styles';
import { Box, Drawer, useTheme } from '@mui/material';
import { SIDEBAR_DRAWER_WIDTH } from '@/constants/constants';
import ScrollBar from '@/components/ScrollBar/ScrollBar';
import NavSection from '@/components/NavSection/NavSection';
import ThemeModeSwitch from '@/components/ThemeModeSwitch/ThemeModeSwitch';
import MHidden from '@/components/@MUI-Extended/MHidden';
import sideBarConfig from './dashboardSideBarItems';
import { IDashboardSidebarOnCloseCallback } from './types';

/**
 * Dashboard Sidebar Root Style
 *
 * Application dashboard layout root component styling
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 *
 * @component
 * @return {Component} - styled dashboard layout root component
 */
const RootStyle = styled('div')(({ theme }) => ({
  [theme.breakpoints.up('lg')]: {
    flexShrink: 0,
    width: SIDEBAR_DRAWER_WIDTH,
  },
}));

/**
 * Dashboard Sidebar Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 *
 * @typedef IDashboardSideBarProps
 * @prop {boolean} isOpenSidebar - sidebar open flag
 * @prop {IDashboardSidebarOnCloseCallback} onCloseSidebar - sidebar close callback
 */
interface IDashboardSideBarProps {
  isOpenSidebar?: boolean;
  onCloseSidebar?: IDashboardSidebarOnCloseCallback;
}

/**
 * Dashboard Sidebar
 *
 * Global dashboard sidebar component
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @component
 * @param {IDashboardSideBarProps} props - component props
 * @returns {FC} - dashboard sidebar component
 */
const DashboardSideBar: FC<IDashboardSideBarProps> = (props) => {
  const { isOpenSidebar, onCloseSidebar } = props;
  const { pathname } = useRouter();
  const theme = useTheme();

  useEffect(() => {
    if (onCloseSidebar) {
      onCloseSidebar();
    }
  }, [pathname, onCloseSidebar]);

  /**
   * Sidebar Content
   *
   * Sidebar content area renderer helper component
   *
   * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
   * @since 0.0.1
   *
   * @component
   * @returns {Component} - sidebar content component
   */
  const renderContent = (
    <ScrollBar
      sx={{
        height: '100%',
        overflowY: 'auto',
        '& .simplebar-content': {
          height: '100%',
          display: 'flex',
          flexDirection: 'column',
        },
        '& .simplebar-placeholder': {
          display: 'none',
        },
      }}
    >
      <Box sx={{ mt: 1, px: 2.5, py: 3 }}>
        <Box component={Link} href="/" sx={{ px: 2, display: 'inline-flex' }}>
          {theme.palette.mode === 'dark' && (
            <Box width={150} component="img" src="/logowhite.webp" />
          )}
          {theme.palette.mode === 'light' && (
            <Box width={150} component="img" src="/logoblack.webp" />
          )}
        </Box>
      </Box>
      <NavSection navConfig={sideBarConfig} />
      <Box display="flex" p={5}>
        <ThemeModeSwitch />
      </Box>
    </ScrollBar>
  );

  return (
    <RootStyle>
      <MHidden width="lgUp">
        <Drawer
          open={isOpenSidebar}
          onClose={onCloseSidebar}
          PaperProps={{
            sx: { width: SIDEBAR_DRAWER_WIDTH },
          }}
        >
          {renderContent}
        </Drawer>
      </MHidden>

      <MHidden width="lgDown">
        <Drawer
          open
          variant="persistent"
          PaperProps={{
            sx: { width: SIDEBAR_DRAWER_WIDTH },
          }}
        >
          {renderContent}
        </Drawer>
      </MHidden>
    </RootStyle>
  );
};

export default DashboardSideBar;
