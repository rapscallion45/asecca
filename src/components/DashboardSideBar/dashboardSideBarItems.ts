import HomeIcon from '@mui/icons-material/Home';
import ViewKanbanIcon from '@mui/icons-material/ViewKanban';
import MonetizationOnIcon from '@mui/icons-material/MonetizationOn';
import ListAltIcon from '@mui/icons-material/ListAlt';
import TableViewIcon from '@mui/icons-material/TableView';
import { INavItemConfig } from '../NavItem/types';

/**
 * Dashboard Sidebar navigation menu item definitions
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 * @memberof DashboardSideBar
 *
 * @constant
 * @type {Array<Partial<INavItemConfig>>}
 */
const dashboardSideBarItems: Array<Partial<INavItemConfig>> = [
  {
    title: 'general',
  },
  {
    title: 'home',
    path: '/',
    icon: HomeIcon,
  },
  {
    title: 'kanban boards',
  },
  {
    title: 'Collections',
    path: '/dashboard/kanban/collection',
    icon: ViewKanbanIcon,
  },
  {
    title: 'Projects',
    path: '/dashboard/kanban/collection-projects',
    icon: ViewKanbanIcon,
  },
  {
    title: 'Configurators',
  },
  {
    title: 'Cost Configurator',
    path: '/dashboard/configurators/costs-config',
    icon: MonetizationOnIcon,
  },
  {
    title: 'Forms',
  },
  {
    title: 'Collection Form',
    path: '/dashboard/forms/collection-form',
    icon: ListAltIcon,
  },
  {
    title: 'Tables',
  },
  {
    title: 'Devices',
    path: '/dashboard/tables/devices',
    icon: TableViewIcon,
  },
  {
    title: 'Table 2',
    path: '/dashboard/tables/aggrid-page-2',
    icon: TableViewIcon,
  },
  {
    title: 'Table 3',
    path: '/dashboard/tables/aggrid-page-3',
    icon: TableViewIcon,
  },
  {
    title: 'Table 4',
    path: '/dashboard/tables/aggrid-page-4',
    icon: TableViewIcon,
  },
];

export default dashboardSideBarItems;
