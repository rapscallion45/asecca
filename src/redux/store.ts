import { configureStore, Action } from '@reduxjs/toolkit';
import { ThunkAction } from 'redux-thunk';
import logger from 'redux-logger';
import userPermissionReducer from './slices/userPermissionSlice';
import costsConfigReducer from './slices/costsConfigSlice';
import notificationsReducer from './slices/notificationsSlice';
import themeReducer from './slices/themeSlice';
import { collectionsKanbanSlice } from './slices/kanbanSlice';

/**
 * Configuration of the redux store to manage the global application state
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @see See [more info on Redux Store object](https://redux.js.org/api/store)
 *
 * @constant
 * @type {ToolkitStore<Object>}
 */
const store = configureStore({
  reducer: {
    userPermission: userPermissionReducer,
    costsConfig: costsConfigReducer,
    notifications: notificationsReducer,
    theme: themeReducer,
    collectionsKanban: collectionsKanbanSlice.reducer,
  },
  // @ts-ignore
  middleware:
    process.env.NODE_ENV === 'development'
      ? (getDefaultMiddleware) => getDefaultMiddleware().concat(logger)
      : (getDefaultMiddleware) => getDefaultMiddleware(),
});

/**
 * Shorthand application store dispatch type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppDispatch - application specific store dispatch type
 */
export type AppDispatch = typeof store.dispatch;

/**
 * Shorthand application state type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppState - application specific state object shape
 */
export type AppState = ReturnType<typeof store.getState>;

/**
 * Shorthand application Thunk Action type helper
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @typedef AppThunk - application specific Thunk action definition
 */
export type AppThunk = ThunkAction<void, AppState, unknown, Action<string>>;

export default store;
