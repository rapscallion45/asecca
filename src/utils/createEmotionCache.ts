import createCache from '@emotion/cache';

/* create emotion style cache, and set 'prepend' true to allow MUI style overrides */

const createEmotionCache = () => createCache({ key: 'css', prepend: true });

export default createEmotionCache;
