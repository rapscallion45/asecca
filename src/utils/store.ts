/* redux store configuration placeholder */
const useStore = () => {};
export default useStore;
