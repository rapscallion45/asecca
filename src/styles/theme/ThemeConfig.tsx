import { FC, useMemo, ReactNode } from 'react';
import { useSelector } from 'react-redux';
import { CacheProvider, EmotionCache } from '@emotion/react';
import { CssBaseline } from '@mui/material';
import {
  ThemeProvider,
  createTheme,
  Theme,
  ThemeOptions,
} from '@mui/material/styles';
import useMediaQuery from '@mui/material/useMediaQuery';
import { AppState } from '@/redux/store';
import lightThemePalette from './lightTheme/lightThemePalette';
import darkThemePalette from './darkTheme/darkThemePalette';
import typography from './typography';

/**
 * Theme Config Props
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @see See [more info on creating emotion styling engine cache](https://emotion.sh/docs/introduction)
 *
 * @typedef IThemeConfigProps
 * @prop {EmotionCache} emotionCache - emotion styling engine cache
 * @prop {ReactNode} children - component child node
 */
interface IThemeConfigProps {
  emotionCache: EmotionCache;
  children?: ReactNode;
}

/**
 * Theme Config
 *
 * Global application theme definitions and setup
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.0
 *
 * @component
 * @param {IThemeConfigProps} props - component props
 * @returns {FC} - theme config functional component
 */
const ThemeConfig: FC<IThemeConfigProps> = (props) => {
  const { emotionCache, children } = props;

  /* grab global theme state */
  const { type: themeType } = useSelector((state: AppState) => state.theme);

  /* check if user has dark mode preference */
  const prefersDarkMode = useMediaQuery('(prefers-color-scheme: dark)');

  /* get options and create theme for current state */
  const themeOptions: ThemeOptions = useMemo(
    () => ({
      palette:
        themeType === 'light' && !prefersDarkMode
          ? lightThemePalette
          : darkThemePalette,
      typography,
    }),
    [themeType, prefersDarkMode]
  );
  const theme: Theme = createTheme(themeOptions);

  return (
    <CacheProvider value={emotionCache}>
      <ThemeProvider theme={theme}>
        <CssBaseline />
        {children}
      </ThemeProvider>
    </CacheProvider>
  );
};

export default ThemeConfig;
