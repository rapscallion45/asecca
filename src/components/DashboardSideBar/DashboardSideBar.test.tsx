import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import DashboardSideBar from './DashboardSideBar';
import ThemeConfig from '../../styles/theme/ThemeConfig';
import createEmotionCache from '../../utils/createEmotionCache';
import store from '../../redux/store';

/**
 * Dashboard SideBar Unit Tests
 *
 * @author Carl Scrivener {@link https://github.com/rapscallion45 GitHub}
 * @since 0.0.1
 */
describe('Dashboard SideBar', () => {
  it('Renders correctly', async () => {
    /** perform snapshot test */
    const tree = renderer
      .create(
        <Provider store={store}>
          <ThemeConfig emotionCache={createEmotionCache()}>
            <DashboardSideBar />
          </ThemeConfig>
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
