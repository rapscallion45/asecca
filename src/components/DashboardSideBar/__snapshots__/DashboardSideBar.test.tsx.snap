// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard SideBar Renders correctly 1`] = `
<div
  className="MuiBox-root css-1j019pi"
>
  <div
    data-simplebar="init"
  >
    <div
      className="simplebar-wrapper"
    >
      <div
        className="simplebar-height-auto-observer-wrapper"
      >
        <div
          className="simplebar-height-auto-observer"
        />
      </div>
      <div
        className="simplebar-mask"
      >
        <div
          className="simplebar-offset"
        >
          <div
            aria-label="scrollable content"
            className="simplebar-content-wrapper"
            role="region"
            tabIndex={0}
          >
            <div
              className="simplebar-content"
            />
          </div>
        </div>
      </div>
      <div
        className="simplebar-placeholder"
      />
    </div>
    <div
      className="simplebar-track simplebar-horizontal"
    >
      <div
        className="simplebar-scrollbar"
      />
    </div>
    <div
      className="simplebar-track simplebar-vertical"
    >
      <div
        className="simplebar-scrollbar"
      />
    </div>
  </div>
  <div
    data-simplebar="init"
  >
    <div
      className="simplebar-wrapper"
    >
      <div
        className="simplebar-height-auto-observer-wrapper"
      >
        <div
          className="simplebar-height-auto-observer"
        />
      </div>
      <div
        className="simplebar-mask"
      >
        <div
          className="simplebar-offset"
        >
          <div
            aria-label="scrollable content"
            className="simplebar-content-wrapper"
            role="region"
            tabIndex={0}
          >
            <div
              className="simplebar-content"
            >
              <div
                className="MuiDrawer-root MuiDrawer-docked css-pw1gwd-MuiDrawer-docked"
              >
                <div
                  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiDrawer-paper MuiDrawer-paperAnchorLeft MuiDrawer-paperAnchorDockedLeft css-1da1smj-MuiPaper-root-MuiDrawer-paper"
                >
                  <div
                    className="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-lp1u1-MuiToolbar-root"
                  >
                    <button
                      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-h3bten-MuiButtonBase-root-MuiIconButton-root"
                      disabled={false}
                      onBlur={[Function]}
                      onClick={[Function]}
                      onContextMenu={[Function]}
                      onDragLeave={[Function]}
                      onFocus={[Function]}
                      onKeyDown={[Function]}
                      onKeyUp={[Function]}
                      onMouseDown={[Function]}
                      onMouseLeave={[Function]}
                      onMouseUp={[Function]}
                      onTouchEnd={[Function]}
                      onTouchMove={[Function]}
                      onTouchStart={[Function]}
                      tabIndex={0}
                      type="button"
                    >
                      <svg
                        aria-hidden={true}
                        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                        data-testid="MenuIcon"
                        focusable="false"
                        viewBox="0 0 24 24"
                      >
                        <path
                          d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
                        />
                      </svg>
                    </button>
                  </div>
                  <hr
                    className="MuiDivider-root MuiDivider-fullWidth css-17ac5dh-MuiDivider-root"
                  />
                  <nav
                    className="MuiList-root MuiList-padding css-h4y409-MuiList-root"
                  >
                    <div
                      className="MuiBox-root css-tm91m2"
                    >
                      <a
                        className="MuiBox-root css-oudrid"
                        href="/"
                        onClick={[Function]}
                        onMouseEnter={[Function]}
                        onTouchStart={[Function]}
                      >
                        <img
                          className="MuiBox-root css-1nbic85"
                          src="/logoblack.webp"
                        />
                      </a>
                    </div>
                    <div
                      className="MuiBox-root css-0"
                    >
                      <ul
                        className="MuiList-root css-1mk9mw3-MuiList-root"
                      >
                        <li
                          className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-1w5xxiy-MuiListSubheader-root"
                        >
                          general
                        </li>
                        <a
                          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-in96y6-MuiButtonBase-root-MuiListItemButton-root"
                          href="/"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          target=""
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yxzc5o-MuiSvgIcon-root"
                            data-testid="HomeIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
                            />
                          </svg>
                          <div
                            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                          >
                            home
                          </div>
                        </a>
                        <li
                          className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-1w5xxiy-MuiListSubheader-root"
                        >
                          kanban boards
                        </li>
                        <a
                          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-in96y6-MuiButtonBase-root-MuiListItemButton-root"
                          href="/dashboard/kanban/collection"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          target=""
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yxzc5o-MuiSvgIcon-root"
                            data-testid="ViewKanbanIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7V7h2v10zm4-5h-2V7h2v5zm4 3h-2V7h2v8z"
                            />
                          </svg>
                          <div
                            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                          >
                            Collections
                          </div>
                        </a>
                        <a
                          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-in96y6-MuiButtonBase-root-MuiListItemButton-root"
                          href="/dashboard/kanban/collection-projects"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          target=""
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yxzc5o-MuiSvgIcon-root"
                            data-testid="ViewKanbanIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 3H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2zM9 17H7V7h2v10zm4-5h-2V7h2v5zm4 3h-2V7h2v8z"
                            />
                          </svg>
                          <div
                            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                          >
                            Projects
                          </div>
                        </a>
                        <li
                          className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-1w5xxiy-MuiListSubheader-root"
                        >
                          Configurators
                        </li>
                        <a
                          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-in96y6-MuiButtonBase-root-MuiListItemButton-root"
                          href="/dashboard/configurators/costs-config"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          target=""
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yxzc5o-MuiSvgIcon-root"
                            data-testid="MonetizationOnIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm1.41 16.09V20h-2.67v-1.93c-1.71-.36-3.16-1.46-3.27-3.4h1.96c.1 1.05.82 1.87 2.65 1.87 1.96 0 2.4-.98 2.4-1.59 0-.83-.44-1.61-2.67-2.14-2.48-.6-4.18-1.62-4.18-3.67 0-1.72 1.39-2.84 3.11-3.21V4h2.67v1.95c1.86.45 2.79 1.86 2.85 3.39H14.3c-.05-1.11-.64-1.87-2.22-1.87-1.5 0-2.4.68-2.4 1.64 0 .84.65 1.39 2.67 1.91s4.18 1.39 4.18 3.91c-.01 1.83-1.38 2.83-3.12 3.16z"
                            />
                          </svg>
                          <div
                            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                          >
                            Cost Configurator
                          </div>
                        </a>
                        <li
                          className="MuiListSubheader-root MuiListSubheader-gutters MuiListSubheader-sticky css-1w5xxiy-MuiListSubheader-root"
                        >
                          Tables
                        </li>
                        <a
                          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-in96y6-MuiButtonBase-root-MuiListItemButton-root"
                          href="/dashboard/tables/devices"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          target=""
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yxzc5o-MuiSvgIcon-root"
                            data-testid="TableViewIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 7H9c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm0 2v2H9V9h10zm-6 6v-2h2v2h-2zm2 2v2h-2v-2h2zm-4-2H9v-2h2v2zm6-2h2v2h-2v-2zm-8 4h2v2H9v-2zm8 2v-2h2v2h-2zM6 17H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2v1h-2V5H5v10h1v2z"
                            />
                          </svg>
                          <div
                            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                          >
                            Devices
                          </div>
                        </a>
                        <a
                          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-in96y6-MuiButtonBase-root-MuiListItemButton-root"
                          href="/dashboard/tables/aggrid-page-2"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          target=""
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yxzc5o-MuiSvgIcon-root"
                            data-testid="TableViewIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 7H9c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm0 2v2H9V9h10zm-6 6v-2h2v2h-2zm2 2v2h-2v-2h2zm-4-2H9v-2h2v2zm6-2h2v2h-2v-2zm-8 4h2v2H9v-2zm8 2v-2h2v2h-2zM6 17H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2v1h-2V5H5v10h1v2z"
                            />
                          </svg>
                          <div
                            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                          >
                            Table 2
                          </div>
                        </a>
                        <a
                          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-in96y6-MuiButtonBase-root-MuiListItemButton-root"
                          href="/dashboard/tables/aggrid-page-3"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          target=""
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yxzc5o-MuiSvgIcon-root"
                            data-testid="TableViewIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 7H9c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm0 2v2H9V9h10zm-6 6v-2h2v2h-2zm2 2v2h-2v-2h2zm-4-2H9v-2h2v2zm6-2h2v2h-2v-2zm-8 4h2v2H9v-2zm8 2v-2h2v2h-2zM6 17H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2v1h-2V5H5v10h1v2z"
                            />
                          </svg>
                          <div
                            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                          >
                            Table 3
                          </div>
                        </a>
                        <a
                          className="MuiButtonBase-root MuiListItemButton-root MuiListItemButton-gutters MuiListItemButton-root MuiListItemButton-gutters css-in96y6-MuiButtonBase-root-MuiListItemButton-root"
                          href="/dashboard/tables/aggrid-page-4"
                          onBlur={[Function]}
                          onClick={[Function]}
                          onContextMenu={[Function]}
                          onDragLeave={[Function]}
                          onFocus={[Function]}
                          onKeyDown={[Function]}
                          onKeyUp={[Function]}
                          onMouseDown={[Function]}
                          onMouseEnter={[Function]}
                          onMouseLeave={[Function]}
                          onMouseUp={[Function]}
                          onTouchEnd={[Function]}
                          onTouchMove={[Function]}
                          onTouchStart={[Function]}
                          tabIndex={0}
                          target=""
                        >
                          <svg
                            aria-hidden={true}
                            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1yxzc5o-MuiSvgIcon-root"
                            data-testid="TableViewIcon"
                            focusable="false"
                            viewBox="0 0 24 24"
                          >
                            <path
                              d="M19 7H9c-1.1 0-2 .9-2 2v10c0 1.1.9 2 2 2h10c1.1 0 2-.9 2-2V9c0-1.1-.9-2-2-2zm0 2v2H9V9h10zm-6 6v-2h2v2h-2zm2 2v2h-2v-2h2zm-4-2H9v-2h2v2zm6-2h2v2h-2v-2zm-8 4h2v2H9v-2zm8 2v-2h2v2h-2zM6 17H5c-1.1 0-2-.9-2-2V5c0-1.1.9-2 2-2h10c1.1 0 2 .9 2 2v1h-2V5H5v10h1v2z"
                            />
                          </svg>
                          <div
                            className="MuiListItemText-root css-tlelie-MuiListItemText-root"
                          >
                            Table 4
                          </div>
                        </a>
                      </ul>
                    </div>
                    <div
                      className="MuiBox-root css-131uyni"
                    >
                      <div
                        className="MuiBox-root css-1tkjpgz"
                      >
                        <label
                          className="MuiFormControlLabel-root MuiFormControlLabel-labelPlacementEnd css-1vbggne-MuiFormControlLabel-root"
                          data-testid="light-dark-btn"
                        >
                          <span
                            className="MuiSwitch-root MuiSwitch-sizeMedium css-sj51q3-MuiSwitch-root"
                          >
                            <span
                              className="MuiButtonBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary PrivateSwitchBase-root MuiSwitch-switchBase MuiSwitch-colorSecondary css-7la7f4-MuiButtonBase-root-MuiSwitch-switchBase"
                              onBlur={[Function]}
                              onContextMenu={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              tabIndex={null}
                            >
                              <input
                                checked={false}
                                className="PrivateSwitchBase-input MuiSwitch-input css-1m9pwf3"
                                onChange={[Function]}
                                required={false}
                                type="checkbox"
                              />
                              <span
                                className="MuiSwitch-thumb css-jsexje-MuiSwitch-thumb"
                              />
                            </span>
                            <span
                              className="MuiSwitch-track css-j4fnnk-MuiSwitch-track"
                            />
                          </span>
                          <span
                            className="MuiTypography-root MuiTypography-body1 MuiFormControlLabel-label css-1lyi4-MuiTypography-root"
                          />
                        </label>
                      </div>
                    </div>
                  </nav>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        className="simplebar-placeholder"
      />
    </div>
    <div
      className="simplebar-track simplebar-horizontal"
    >
      <div
        className="simplebar-scrollbar"
      />
    </div>
    <div
      className="simplebar-track simplebar-vertical"
    >
      <div
        className="simplebar-scrollbar"
      />
    </div>
  </div>
</div>
`;
